aws/resource_aws_sns_sms_preferences.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSnsSmsPreferences() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSnsSmsPreferencesSet,
+		Read:   resourceAwsSnsSmsPreferencesGet,
+		Update: resourceAwsSnsSmsPreferencesSet,
+		Delete: resourceAwsSnsSmsPreferencesDelete,
+
+		Schema: map[string]*schema.Schema{
+			"monthly_spend_limit": {
+				Type:     schema.TypeString,
```

Comment:
This is tricky because the AWS API treats all those attributes as Strings, including monthly spend limit and delivery sampling rate, and an empty string "" (not defined) is not the same as a "0" (explicit 0).
Using int values in TF won't allow us to distinguish between the 2 cases and will always set those values as "0" when retrieved as "", which would be incorrect.